### PR TITLE
setup telemetry and add account_connected event

### PR DESCRIPTION
**Pull request by trilopin**
#### Summary
Setup telemetry for ServiceNow.
Track account_connected event when the user succeeds to connect their ServiceNow account.

server/telemetry has been mostly copied (with some tweaks) from [pluginapi](https://github.com/mattermost/mattermost-plugin-api) package in order to avoid being forced to upgrade to mattermost-server/v6.

A couple of opportunistic improvements done in Makefile to speed-up builds while developing.

#### Ticket Link
FIxes https://mattermost.atlassian.net/browse/MM-51748

### CODE REVIEW

File: build/custom.mk [hunk 1]
```diff
@@ -1 +1,9 @@
 # Include custom targets and environment variables here
+
+# If there's no MM_RUDDER_PLUGINS_PROD, add DEV data
+RUDDER_WRITE_KEY = 1d5bMvdrfWClLxgK1FvV3s4U1tg
+ifdef MM_RUDDER_PLUGINS_PROD
+	RUDDER_WRITE_KEY = $(MM_RUDDER_PLUGINS_PROD)
+endif
+
+GO_BUILD_FLAGS += -ldflags '-X "github.com/mattermost/mattermost-plugin-servicenow/telemetry.rudderWriteKey=$(RUDDER_WRITE_KEY)"'
```
Comment (trilopin) [inline build/custom.mk:9]: this is addressed locally instead through plugin-api/experimental/telemetry to avoid migration to v6

**Review by mickmister — APPROVED** [review]

**Review by hanzei — COMMENTED** [review]
https://github.com/mattermost/mattermost-plugin-servicenow/pull/178 will migrate the codebase to use v6. If we wait for that PR to be merged first, we don't need to duplicate code into this repo.

@trilopin Can we wait for that PR to be merged?

**Comment by trilopin** [comment]
> @trilopin Can we wait for that PR to be merged?

yes, no prob

File: go.mod [hunk 2]
```diff
@@ -5,9 +5,9 @@ go 1.16
 require (
 	bou.ke/monkey v1.0.2
 	github.com/gorilla/mux v1.8.0
-	github.com/mattermost/mattermost-plugin-api v0.0.27
-	github.com/mattermost/mattermost-server/v6 v6.3.0
+	github.com/mattermost/mattermost-plugin-api v0.1.3
+	github.com/mattermost/mattermost-server/v6 v6.0.0-20221109191448-21aec2741bfe
```
Comment (trilopin) [inline go.mod:9]: needed to do this to avoid opengraph ambiguous import (same pinned commit as github plugin), should I expect any side effect from that?

Comment (trilopin) [reply go.mod:9]: min_server_version is 7.1.0 at github BTW

**Comment by trilopin** [comment]
closed in favor of https://github.com/mattermost/mattermost-plugin-servicenow/pull/183, where we protect min_server_version by avoiding pluginapi update.

**Pull request closed without merging**